inventory/services_test.go

```diff
@@ -0,0 +1,344 @@
+		defer removeServices(t, serviceID)
+
+		service, err := client.Default.Services.GetService(&services.GetServiceParams{
+			Body:    services.GetServiceBody{ServiceID: serviceID},
+			Context: context.TODO(),
+		})
+		require.NoError(t, err)
+
+		// Change MySQL service name.
+		newPort := int64(3337)
+		changeRes, err := client.Default.Services.ChangeMySQLService(&services.ChangeMySQLServiceParams{
+			Body: services.ChangeMySQLServiceBody{
+				ServiceID: serviceID,
+				Port:      newPort,
+			},
+			Context: context.TODO(),
+		})
+		require.NoError(t, err)
+		require.NotNil(t, changeRes.Payload.Mysql)
+		require.Equal(t, serviceID, changeRes.Payload.Mysql.ServiceID)
+		require.Equal(t, newPort, changeRes.Payload.Mysql.Port)
+
+		// Check changes in backend.
+		changedService, err := client.Default.Services.GetService(&services.GetServiceParams{
+			Body:    services.GetServiceBody{ServiceID: serviceID},
+			Context: context.TODO(),
+		})
+		require.NoError(t, err)
+		require.Equal(t, newPort, changedService.Payload.Mysql.Port)
+		// Check that other fields isn't changed.
+		require.Equal(t, service.Payload.Mysql.ServiceName, changedService.Payload.Mysql.ServiceName)
+		require.Equal(t, service.Payload.Mysql.Address, changedService.Payload.Mysql.Address)
+		require.Equal(t, service.Payload.Mysql.NodeID, changedService.Payload.Mysql.NodeID)
+	})
+
+	t.Run("AddNodeIDEmpty", func(t *testing.T) {
+		t.Parallel()
+		params := &services.AddMySQLServiceParams{
+			Body: services.AddMySQLServiceBody{
+				NodeID:      "",
+				Address:     "localhost",
+				Port:        3306,
+				ServiceName: "MySQL Service with empty node id",
+			},
+			Context: context.TODO(),
+		}
+		res, err := client.Default.Services.AddMySQLService(params)
+		require.Error(t, err) // Can't use EqualError because it returns different references each time.
+		require.Contains(t, err.Error(), "unknown error (status 400)")
+		require.Nil(t, res)
+	})
+}
+
+func TestAmazonRDSMySQLService(t *testing.T) {
+	t.Skip("Not implemented yet.")
+	t.Run("Basic", func(t *testing.T) {
+		params := &services.AddAmazonRDSMySQLServiceParams{
+			Body: services.AddAmazonRDSMySQLServiceBody{
+				NodeID:      "pmm-server",
+				Address:     "localhost",
+				Port:        3306,
+				ServiceName: "Basic AmazonRDSMySQL Service",
+			},
+			Context: context.TODO(),
+		}
+		res, err := client.Default.Services.AddAmazonRDSMySQLService(params)
+		require.NoError(t, err)
+		require.NotNil(t, res)
+		require.NotNil(t, res.Payload.AmazonRDSMysql)
+		require.NotEmpty(t, res.Payload.AmazonRDSMysql.ServiceID)
+		serviceID := res.Payload.AmazonRDSMysql.ServiceID
+		defer removeServices(t, serviceID)
+		require.Equal(t, "pmm-server", res.Payload.AmazonRDSMysql.NodeID)
+		require.Equal(t, "localhost", res.Payload.AmazonRDSMysql.Address)
+		require.Equal(t, int64(3306), res.Payload.AmazonRDSMysql.Port)
+		require.Equal(t, "Basic AmazonRDSMySQL Service", res.Payload.AmazonRDSMysql.ServiceName)
```

Comment:
Two notes in general:
1. Prefer `assert` to `require`. The latter one should be used when we can't continue with that test. For example, if function `func foo() (*Result, error)` returns error, we can't continue. But then we should check `*Result` with `assert`.
2. Checking field-by-filed with `require` does not give us enough information to fix the tests – it stops too early. Compare the whole struct results after zeroing fields that are new everytime (like UUIDs).

For example (in pseudocode):
```go
res, err := client.Default.Services.AddAmazonRDSMySQLService(params)
require.NoError(t, err)

expected := AmazonRDSMysqlType{
	NodeID: "pmm-server",
	Address: "localhost",
	Port: 3306,
	ServiceName: "Basic AmazonRDSMySQL Service",
}
assert.Equal(t, expected, res.Payload.AmazonRDSMysql)
```

Reply:
ok